recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class SdbusCppConan(ConanFile):
+    name = "sdbus-cpp"
+    license = "LGPL-2.1-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Kistler-Group/sdbus-cpp"
+    description = "High-level C++ D-Bus library for Linux designed" \
+                  " to provide easy-to-use yet powerful API in modern C++"
+    topics = ("conan", "dbus", "sd-bus", "sdbus-c++", "sdbus-cpp")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_code_gen": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_code_gen": False,
+    }
+    exports_sources = "CMakeLists.txt"
+    generators = ("cmake", "pkg_config")
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "gcc": "7",
+            "clang": "6",
+        }
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
+        if self.options.shared:
+            del self.options.fPIC
```

Comment:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
    
    def validate(self):
        if self.settings.os != "Linux":
            raise ConanInvalidConfiguration("Only Linux supported")
```

https://blog.conan.io/2020/12/14/New-conan-release-1-32.html